example_test.go

```diff
@@ -0,0 +1,30 @@
+package datadog
+
+import (
+	"github.com/DataDog/datadog-go/statsd"
+	"go.opentelemetry.io/otel/api/global"
+	export "go.opentelemetry.io/otel/sdk/export/metric"
+	"go.opentelemetry.io/otel/sdk/metric/batcher/ungrouped"
+	"go.opentelemetry.io/otel/sdk/metric/controller/push"
+)
+
+func ExampleExporter() {
+	exporter, _ := NewExporter(
+		Config{
+			StatsAddr:       DefaultStatsAddrUDP,
+			UseDistribution: false,
+		},
+		statsd.WithoutTelemetry(),
+		statsd.WithNamespace("crl_"),
+		statsd.WithTags([]string{
+			"example_tag:example_value",
+		}),
+	)
+
+	selector := NewWithDataDogMeasure()
+	batcher := ungrouped.New(selector, export.NewDefaultLabelEncoder(), false)
+	pusher := push.New(batcher, exporter,1)
```

Comment:
I believe the 1 here, really should be plumbed into the exporter as well, as its currently hard coded as a const of 1, when the rate's reported may not be 1 if a larger flush interval is set.

Reply:
AFAIK, the rate const in the exporter is for sampling rate, not flush rate. 

As for this `1` here, this should be left up to the user. If you use something like `ddsketch` you can be a lot more forgiving and have your interval flush every 10 seconds for example

Reply:
Yeah. This raises an interesting scenario though where we allow two flush intervals to be set, that of the OTel client, and that which belongs to the buffered DD client. I wonder if we shouldn't expose directly all settings of the DD client, and only allow users to select those that make sense.

Reply:
I needed to leave it open for specifically  testing just that. The UDP socket still needs to buffer and flush, no way around that and will always need to expose it imo for high volume tuning or special networks.

Reply:
@zlozano for the same reason I prefer `ddsketch` over `exactMeasure` (and for the same reason I prefer having the batcher/aggregator export the values directly instead of doing 1 second exactMeasures), you would want a set up that doesn't care about what the amount of the flush interval is. 

In other words, if you changed the flush interval from 1 second to 10 seconds, the correctness of your application should remain the same, and the pressure on your memory/cpu should remain the same as well. 

I definitely understand that the combination of `exactMeasure` with synchronizing quick flushes (Otel client + DD client) is a valid work around for now, but imho it's hard to maintain and debug :) 

> I wonder if we shouldn't expose directly all settings of the DD client, and only allow users to select those that make sense.

You can certainly block certain options, or you can write some documentation and trust engineers will follow it 😄🙂

Reply:
> The UDP socket still needs to buffer and flush, no way around that 

💯 I don't disagree there.  The datadog buffer is needed here to ensure we don't exceed the optimal UDP message size. My question is not around buffering itself, but the intervals on which we flush the buffers.

I guess it's the tuning part I am having trouble wrapping my head around. What would be a use case where configuring the OTel pusher interval _and_ the datadog flush interval would be advantageous? The previous exporter implementation  (v0.2.3) flushed the exporter immediately.

Reply:
@marwan-at-work maybe a quick offline sync would help me better understand :)

> In other words, if you changed the flush interval from 1 second to 10 seconds, the correctness of your application should remain the same, and the pressure on your memory/cpu should remain the same as well.

I don't doubt the usefulness of being able to control how often metrics get pushed to a collector. My question is around why do we want to expose two ways of doing that. Apologies if I misunderstood your comment.

Reply:
This is all non-blocking BTW. I am just trying to better understand exposing some settings that seem to overlap with OTel settings. I think what I am unclear on is 1) is that in fact what's happening and 2) if so, how do we plan to help consumers know when to configure what value.

Reply:
Ya I’m not sure yet. Like I mentioned, I just wanted a version with it exposed so I could do exploratory testing. I’m also curious whether the client can/should be instantiated by the implementer to aid running existing instrumentation outside of otel using the same client. Thus allowing a much easier time to port metrics gradually?  Either way, there is much understand here to ensure accuracy and type safety of the exported data.